# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaManager.h
# Commit: 2fd502bdc2d9
# Full Hash: 2fd502bdc2d921b888fc6597b6b3d3514226e8a2
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-12 20:37:56
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Update the cached value in PersistOp::DoDirectoryWork only after successful creation of the metadata file; r=dom-storage-reviewers,jari
#   
#   Currently, the cached value is updated before creation of the metadata file
#   which can lead to a mismatch between the metadata file and the cached value if
#   creation of the metadata file fails. It will be safer to update the cached
# ==============================================================================

diff -r 8b98c798b362 -r 2fd502bdc2d9 dom/quota/QuotaManager.h
--- a/dom/quota/QuotaManager.h	Sat Oct 12 10:00:14 2024 +0000
+++ b/dom/quota/QuotaManager.h	Sat Oct 12 10:00:14 2024 +0000
@@ -187,8 +187,7 @@
    *
    * @return last access time of the origin.
    */
-  int64_t NoteOriginDirectoryCreated(const OriginMetadata& aOriginMetadata,
-                                     bool aPersisted);
+  int64_t NoteOriginDirectoryCreated(const OriginMetadata& aOriginMetadata);
 
   // XXX clients can use QuotaObject instead of calling this method directly.
   void DecreaseUsageForClient(const ClientMetadata& aClientMetadata,